### ISSUE TITLE
Redundant check in the code

### ISSUE THREAD

**Issue by kovvalsky** [ParallelMeaningBank/easyccg#6]
If `left.getLeft().isFunctor()` is true than `left.getLeft().isNounOrNP()` is false, right?

https://github.com/ParallelMeaningBank/easyccg/blob/ad2fad18bbfa6d9aa2f01d15d621a4385ecf2b02/src/uk/ac/ed/easyccg/syntax/Combinator.java#L518-L528

**Comment by texttheater** [comment]
True. Seems this was copy and pasted from `BackwardComposition` – here, in the case of `GeneralizedBackwardComposition`, `left` should probably be changed to `leftLeft` on line 524. But would have to find the rule restriction in Steedman (2000) to be completely sure. :)
